src/renderer/dx/DxRenderer.cpp

```diff
@@ -1057,21 +976,38 @@ void DxEngine::_InvalidOr(RECT rc) noexcept
 
         if (SUCCEEDED(hr))
         {
-            if (_invalidScroll.cy != 0 || _invalidScroll.cx != 0)
+            if (_invalidScroll != til::point{ 0, 0 })
             {
-                _presentDirty = _invalidRect;
+                // Copy `til::rectangles` into RECT map.
+                _presentDirty.assign(_invalidMap.begin(), _invalidMap.end());
+
+                // Scale all dirty rectangles into pixels
+                std::transform(_presentDirty.begin(), _presentDirty.end(), _presentDirty.begin(), [&](til::rectangle rc) {
+                    return rc * _glyphCell;

```

Comment:
if `_glyphCell` isn't scaled for DPI then this might be the spot that's causing the artifacts

Reply:
Good call, I'll look.

Reply:
This was one of em.